homeassistant/components/epic_games_store/coordinator.py

```diff
@@ -0,0 +1,188 @@
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN
+
+SCAN_INTERVAL = timedelta(days=1)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EGSUpdateCoordinator(DataUpdateCoordinator[dict[str, list[dict[str, Any]]]]):
+    """Class to manage fetching data from the Epic Game Store."""
+
+    def __init__(
+        self, hass: HomeAssistant, api: EpicGamesStoreAPI, locale: str
+    ) -> None:
+        """Initialize."""
+        self._api = api
+        self.locale = locale
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=SCAN_INTERVAL,
+        )
+
+    async def _async_update_data(self) -> dict[str, list[dict[str, Any]]]:
+        """Update data via library."""
+        try:
+            data = await self.hass.async_add_executor_job(self._api.get_free_games)
+        except Exception as error:
+            raise UpdateFailed(error) from error
+
+        _LOGGER.error(data)
+
+        discount_games = filter(
+            lambda game:
+            # Only accessible from the Store, without a voucher code (effectiveDate is also much in the future "2099-01-01T00:00:00.000Z")
+            not game["isCodeRedemptionOnly"]
+            # Seen some null promotions even with isCodeRedemptionOnly to false (why ?)
+            and game.get("promotions")
+            and (
+                # Current discount(s)
+                game["promotions"]["promotionalOffers"]
+                or
+                # Upcoming discount(s)
+                game["promotions"]["upcomingPromotionalOffers"]
+            ),
+            data["data"]["Catalog"]["searchStore"]["elements"],
+        )
+
+        # free_games = filter(
+        #     lambda game: (
+        #         (
+        #             # Current free game(s)
+        #             game["promotions"]["promotionalOffers"][0]["promotionalOffers"][0][
+        #                 "discountSetting"
+        #             ]["discountPercentage"]
+        #             == 0
+        #             and
+        #             # Checking current price, maybe not necessary
+        #             game["price"]["totalPrice"]["discountPrice"] == 0
+        #         )
+        #         or (
+        #             # Upcoming free game(s)
+        #             game["promotions"]["upcomingPromotionalOffers"][0][
+        #                 "promotionalOffers"
+        #             ][0]["discountSetting"]["discountPercentage"]
+        #             == 0
+        #         )
+        #     ),
+        #     list(discount_games),
+        # )
+
+        _LOGGER.debug(discount_games)
+        # _LOGGER.info(len(data["data"]["Catalog"]["searchStore"]["elements"]))
+        # _LOGGER.info(len(list(discount_games)))
+        # _LOGGER.info(len(list(free_games)))
+
+        # OLD sensor like data
+        return_data: dict[str, list[dict[str, Any]]] = self.data or {}
```

Comment:
At this point I would almost suggest creating your own lib or creating a wrapper around the epic games one